### DISCUSSION TITLE
Correct Segger Embedded Studio settings to get TU_STRCAT3 to work

### DISCUSSION THREAD

**Discussion by Eqqm4n** [hathach/tinyusb#2535] [Q&A] [answered]
Hello-
Currently we have the tinyusb stack working on an STM32F411 using the MAX3421 driver.  The development platform is Segger Embedded Studio.  Debug logging is enabled and by default SES routes this output to its own console window in the IDE.  However it looks like the string concatenation macros like TU_STRCAT3 are producing invalid output (for sure, all these 2X/4X should be valid base 16 numbers):
```
Get Configuration[0] Descriptor
[1:1] Get Descriptor: 2X 2X 2X 2X 2X 2X 2X 2X 
on EP 2X with 0 bytes: P
on EP 2X with 128 bytes: P
[1:1] Control data:
  4X:  2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X  |	.K.....2.......|
  4X:  2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X  |.	.........$....|
  4X:  2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X  |$....$...$......|
  4X:  2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X  |....	...........|
  4X:  2X 2X 2X 2X 2X 2X 2X 2X 2X 2X 2X                 |.@......@..|
on EP 2X with 0 bytes: 

Set Configuration = 1
[1:1] Set Configuration: 2X 2X 2X 2X 2X 2X 2X 2X 
on EP 2X with 0 bytes: P
on EP 2X with 128 bytes: 
[1] Aborted transfer on EP 2X
```
In the project settings the C standard has been adjusted up to gnu17, but this did not help.  It should be noted that the chosen compiler is Segger's, not gcc, and we would like to keep it that way if possible.  Any suggestions?

**Comment by HiFiPhile** [comment, marked as answer]
Looks like you need to change the printf formatter.
In IAR there is a setting, maybe you need to ask Segger.
![image](https://github.com/hathach/tinyusb/assets/4375114/38880a3f-4c13-4ac9-a78a-36b685c7daff)

**Reply by Eqqm4n** [reply]
Hello-
This was correct, the problem was resolved after enabling printf width/precision support.  BTW- what is the intended meaning of the output between the pipes | |?  Is this supposed to be the ASCII versions of the data in the case of strings being returned?

**Reply by HiFiPhile** [reply]
>  Is this supposed to be the ASCII versions of the data in the case of strings being returned?

Yes
